### PR TITLE
Low-hanging nullable fruit

### PR DESCRIPTION
**Pull request by 50Wliu**
There's a bunch of low-hanging fruit to get variables properly classified as nullable (and null-checked).

This pass gets most of src/features strict-clean, with the exception of server.ts and anything that depends on IHostExecutableResolver.

I'm hoping most of the changes should be self-explanatory, let me know if anything looks suspicious. Any changes to functionality are unintentional.

Also let me know if you'd prefer not to receive these types of PRs, because I get that they change a lot of files and reviewing is probably a bit of a small headache.

### CODE REVIEW

File: src/omnisharp/OmnisharpManager.ts [hunk 1]
```diff
@@ -22,10 +22,10 @@ export class OmnisharpManager {
         private platformInfo: PlatformInformation) {
     }
 
-    public async GetOmniSharpLaunchInfo(defaultOmnisharpVersion: string, omnisharpPath: string, useFramework: boolean, serverUrl: string, latestVersionFileServerPath: string, installPath: string, extensionPath: string): Promise<LaunchInfo> {
-        if (!omnisharpPath) {
+    public async GetOmniSharpLaunchInfo(defaultOmnisharpVersion: string, omnisharpPath: string | undefined, useFramework: boolean, serverUrl: string, latestVersionFileServerPath: string, installPath: string, extensionPath: string): Promise<LaunchInfo> {
+        if (omnisharpPath === undefined) {
```
Comment (JoeRobich) [inline src/omnisharp/OmnisharpManager.ts:26]:
```suggestion
        if (omnisharpPath === undefined || omnisharpPath.length === 0) {
```

**Review by JoeRobich — APPROVED** [review]
Thanks for these changes. Enabling more of the TS compilers strict checking is certainly a goal for us. As well as moving from tslint to eslint.

There are a few failing unit tests that have checks like `expect(options.path).to.be.null` that will need to be updated.

File: test/unitTests/OmnisharpManager.test.ts [hunk 2]
```diff
@@ -136,7 +136,7 @@ suite(OmnisharpManager.name, () => {
             });
 
             test('Returns the default path if the omnisharp path is not set', async () => {
-                let launchInfo = await manager.GetOmniSharpLaunchInfo(defaultVersion, "", useFramework, server.baseUrl, latestfilePath, installPath, extensionPath);
+                const launchInfo = await manager.GetOmniSharpLaunchInfo(defaultVersion, undefined, useFramework, server.baseUrl, latestfilePath, installPath, extensionPath);
```
Comment (JoeRobich) [inline test/unitTests/OmnisharpManager.test.ts:139]: I think we should add an additional test here for cases when is set to empty string. Users are very likely to leave `omnisharp.path: ""` in their settings.json when trying to switch from "latest" back to release and the empty string case should be treated like `undefined`.

File: src/omnisharp/OmnisharpManager.ts (hunk 1 above)
Comment (50Wliu) [reply src/omnisharp/OmnisharpManager.ts]: Good catch, you're right. The integration tests also exposed a related configuration problem: the defaults for string-related options are all set to `null` at the moment. Going to push up a test commit so I can run the tests using a Codespace that actually has .NET installed.

Comment (50Wliu) [reply src/omnisharp/OmnisharpManager.ts]: Right, it's because VS Code coerces undefined values for known settings into preset default values: https://github.com/microsoft/vscode/blob/6f26fa19db7e4cd2af57f52fd7009e715c3d7aab/src/vs/platform/configuration/common/configurationRegistry.ts#L578

Of course, this isn't documented anywhere I could find in their API documentation. :)

File: src/omnisharp/options.ts [hunk 3]
```diff
@@ -207,53 +209,44 @@ export class Options {
...
+        function toUseGlobalMonoValue(value?: boolean): string | undefined {
+            if (value === undefined) {
+                return undefined;
+            }
+
             // True means 'always' and false means 'auto'.
             return value ? "always" : "auto";
         }
 
-        if (omnisharpConfig.has('useGlobalMono')) {
```
Comment (50Wliu) [inline src/omnisharp/options.ts:242]: I'm not confident this fallback logic ever worked...I'm running the `singleCsproj` integration test suite right now which doesn't have `useGlobalMono` defined, but this still returns true (with `get` returning "auto", the default).

I suppose the upside is this whole section is going away soon, so I don't suppose it matters too much.

**Review by 50Wliu — COMMENTED** [review]
Ready for review again. Changed all the string-related settings to default to the empty string rather than having to deal with TypeScript warning that it _could_ be undefined. The only time it could be undefined is if the config is removed from package.json, which I think is highly unlikely without corresponding code changes...

File: src/omnisharp/options.ts (hunk 3 above)
Comment (JoeRobich) [reply src/omnisharp/options.ts:242]: This bit of code will be removed prior to the next release to address https://github.com/OmniSharp/omnisharp-vscode/issues/5120

**Review by JoeRobich — APPROVED** [review]
Latest round of changes look good. Thanks!
